darkdetect/_mac_detect.py

```diff
@@ -69,6 +82,37 @@ def isDark():
 def isLight():
     return theme() == 'Light'
 
-#def listener(callback: typing.Callable[[str], None]) -> None:
-def listener(callback):
-    raise NotImplementedError()
+
+def _listen_child():
+    """
+    Run by a child process, install an observer and print theme on change
+    """
+    class Observer(NSObject):
+        def callback_(self, _):
+            try:
+                print(theme(), flush=True)
+            except IOError:
+                os._exit(1)
+    observer = Observer.new()  # Keep a reference of the observer to keep it alive
+    NSDistributedNotificationCenter.defaultCenter().addObserver_selector_name_object_(
+        observer,  # Observer must be kept alive by a different reference
+        "callback:",
+        "AppleInterfaceThemeChangedNotification",
+        None
+    )
+    AppHelper.runConsoleEventLoop()
+
+
+def listener(callback: Callable[[str], None]) -> None:
+    if not _can_listen:
+        raise NotImplementedError()
+    sig = "import signal as s; s.signal(s.SIGINT, s.SIG_IGN)"
+    listen = "import darkdetect as dd; dd._mac_detect._listen_child()"
```

Comment:
In this case, I would avoid importing `darkdetect` as this could use a system-installed copy. As suggested before, I would import `_mac_detect` directly and set `cwd=Path(__file__).parent` to avoid this potential ambiguity.

Reply:
I see, I misunderstood you earlier. I'll try that!